### DISCUSSION TITLE
External type imports working with app.vue but throwing errors with '/pages' directory

### DISCUSSION THREAD

**Discussion by DamianGlowala** [nuxt/nuxt#17410] [Questions] [answered]
I have a multi repo setup:

- /web-app-backend
- /web-app-frontend

Having the following import in one of the components or pages:

`import { SampleEnum } from "@backend/modules/db/sample-module/types/SampleEnum.enum";`

... and the following `nuxt.config`:

```
export default defineNuxtConfig({
    typescript: {
        strict: true,
        typeCheck: true
    },
    alias: {
        "@backend": "../web-app-backend/src"
    }
});
```

... the types are recognized correctly, but the actual imports are not included in the dev bundle:

![image](https://user-images.githubusercontent.com/48835293/174586282-0b2df27e-d936-460e-969a-b7be49eaf428.png)

What should I add, change or delete in the above config to make things work?

On top of the above, is there any difference in the resulting changes the following two make to Nuxt and VS Code's type interference:

- `nuxt.config`'s alias property
- `nuxt.config`'s `typescript.tsConfig.compilerOptions.paths` property?

**Comment by DamianGlowala** [comment]
With only the `alias` property left (as shown below), I am now getting Vue Router warnings in the terminal.
```
export default defineNuxtConfig({
    typescript: {
        strict: true,
        typeCheck: true
    },
    alias: {
        "@backend": "../web-app-backend/src"
    }
});
```

```
[Vue Router warn]: No match found for location with path "/web-app-backend/src/modules/db/contacts/types/ContactCanBeContacted"
[Vue Router warn]: No match found for location with path "/web-app-backend/src/modules/db/fse-schools/types/FseSchoolStatus"
(...)
```

@danielroe, would you have any idea about the above? 🙏 This is currently a blocker in Nuxt 3 for me and I am a bit helpless :/ Any help would be truly appreciated.

**Reply by danielroe** [reply]
Would you create a minimal reproduction?

Note, you should not need to add a `myalias/*` entry.

**Reply by DamianGlowala** [reply]
I certainly will, should be ready within an hour or two. Thanks!

**Reply by DamianGlowala** [reply]
@danielroe here is a very simple reproduction repo.

https://github.com/DamianGlowala/nuxt-framework-discussion-5531-repro

What is worth adding, is that with no `/pages` folder (only `app.vue`), it works beautifully - both the dev mode and production build are fine. Once it is present and imports are located anywhere withing `/pages`, Vue Router starts disliking these. You will see that in the terminal or DevTools.

You can run the `web-app-backend` (a simple Nest.js app) with the following command once the dependencies are installed:
`npm run start:dev`

Let me know if you managed to successfully replicate this behavior on your side, as it does look like a bug to me.

**Comment by DamianGlowala** [comment, marked as answer]
🎉 Made it work with the following `nuxt.config`:

```
import { defineNuxtConfig } from "nuxt";
import { fileURLToPath } from "node:url";

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
    typescript: {
        strict: true,
        typeCheck: true,
        tsConfig: {
            include: ["../../web-app-backend/src/**/*"]
        }
    },
    alias: {
        "@backend": "../web-app-backend/src"
    },
    vite: {
        resolve: {
            alias: {
                "@backend": fileURLToPath(
                    new URL("../web-app-backend/src", import.meta.url)
                )
            }
        }
    }
});
```
